src-java/flowmonitoring-topology/flowmonitoring-storm-topology/src/main/java/org/openkilda/wfm/topology/flowmonitoring/service/ActionService.java

```diff
@@ -99,12 +115,27 @@ public void processFlowLatencyMeasurement(String flowId, FlowDirection direction
         FsmKey key = getFsmKey(flowId, direction);
         FlowLatencyMonitoringFsm fsm = fsms.get(key);
         if (fsm == null) {
-            Flow flow = flowRepository.findById(flowId)
-                    .orElseThrow(() -> new IllegalStateException(format("Flow %s not found.", flowId)));
-            long maxLatency = flow.getMaxLatency() == null || flow.getMaxLatency() == 0
-                    ? Long.MAX_VALUE : flow.getMaxLatency();
-            long maxLatencyTier2 = flow.getMaxLatencyTier2() == null || flow.getMaxLatencyTier2() == 0
-                    ? Long.MAX_VALUE : flow.getMaxLatencyTier2();
+            long maxLatency;
+            long maxLatencyTier2;
+            if (flowRepository.findById(flowId).isPresent()) {
+                Flow flow = flowRepository.findById(flowId).get();
+                maxLatency = flow.getMaxLatency() == null || flow.getMaxLatency() == 0
+                        ? Long.MAX_VALUE : flow.getMaxLatency();
+                maxLatencyTier2 = flow.getMaxLatencyTier2() == null || flow.getMaxLatencyTier2() == 0
+                        ? Long.MAX_VALUE : flow.getMaxLatencyTier2();
+            } else if (haSubFlowRepository.findById(flowId).isPresent()) {
+                HaFlow haFlow = haSubFlowRepository.findById(flowId).get().getHaFlow();
+                if (haFlow != null) {
+                    maxLatency = haFlow.getMaxLatency() == null || haFlow.getMaxLatency() == 0
+                            ? Long.MAX_VALUE : haFlow.getMaxLatency();
+                    maxLatencyTier2 = haFlow.getMaxLatencyTier2() == null || haFlow.getMaxLatencyTier2() == 0
+                            ? Long.MAX_VALUE : haFlow.getMaxLatencyTier2();
```

Comment:
I would suggest to extract the method from this peace of code:
```
flow.getMaxLatency() == null || flow.getMaxLatency() == 0
                        ? Long.MAX_VALUE : flow.getMaxLatency();
```




The extracted method could be like this one:
   ```
 private static long resolveMaxLatency(Long maxLatency) {
        return maxLatency == null || maxLatency == 0
                ? Long.MAX_VALUE : maxLatency;
    }
```


```suggestion
                   Flow flow = flowRepository.findById(flowId).get();
                maxLatency = resolveMaxLatency(flow.getMaxLatency());
                maxLatencyTier2 = resolveMaxLatency(flow.getMaxLatencyTier2());
            } else if (haSubFlowRepository.findById(flowId).isPresent()) {
                HaFlow haFlow = haSubFlowRepository.findById(flowId).get().getHaFlow();
                if (haFlow != null) {
                    maxLatency = resolveMaxLatency(haFlow.getMaxLatency());
                    maxLatencyTier2 = resolveMaxLatency(haFlow.getMaxLatencyTier2());
```

Reply:
It is a great idea. Updated. Thank you!